processing/src/main/java/org/apache/druid/segment/AutoTypeColumnMerger.java

```diff
@@ -120,11 +120,11 @@ public void writeMergedValueDictionary(List<IndexableAdapter> adapters) throws I
             adapter.getNestedColumnMergeables(name)
         );
         final SortedValueDictionary dimValues = mergable.getValueDictionary();
-        mergable.mergeFieldsInto(mergedFields);
 
         boolean allNulls = dimValues == null || dimValues.allNull();
         sortedLookup = dimValues;
         if (!allNulls) {
+          mergable.mergeFieldsInto(mergedFields);
```

Comment:
Just double checking, I had been thinking this was correct outside of the `if` statement because, even if it's all nulls, the others might have never seen a null and so you want to merge it so that null is properly registered.  Is that not a concern?

Reply:
i realized that if it is all null this isn't doing anything useful anyway so i put it inside the if